extensions/amp-iframe/1.0/amp-iframe.js

```diff
@@ -31,6 +33,36 @@ class AmpIframe extends BaseElement {
     );
     return super.isLayoutSupported(layout);
   }
+
+  /** @override */
+  init() {
+    return dict({
+      'onLoadCallback': () => {
+        const placeholder = this.getPlaceholder();
+        if (placeholder) {
+          this.togglePlaceholder(false);
+        } else {
+          measureIntersection(this.element).then((intersectionEntry) => {
+            const {top} = intersectionEntry.boundingClientRect;
+            const viewportHeight = intersectionEntry.rootBounds.height;
+            const minTop = Math.min(600, viewportHeight * 0.75);
+            userAssert(
+              top >= minTop,
+              '<amp-iframe> elements must be positioned outside the first 75% ' +
+                'of the viewport or 600px from the top (whichever is smaller): %s ' +
+                ' Current position %s. Min: %s' +
+                "Positioning rules don't apply for iframes that use `placeholder`." +
+                'See https://github.com/ampproject/amphtml/blob/main/extensions/' +
+                'amp-iframe/amp-iframe.md#iframe-with-placeholder for details.',
+              this.element,
+              top,
+              minTop
+            );
+          });
```

Comment:
(nit: Feel free to forgo this change, but add a TODO)

Let's share this policy across component versions.

```suggestion
          measureIntersection(this.element).then((intersectionEntry) => {
            assertIframeCompliesWithPositionPolicy(
              intersectionEntry.boundingClientRect,
              intersectionEntry.rootBounds
            );
          });
```

Reply:
Added TODO